Plugins/NativeCamera/Source/Android/NativeCameraImpl.cpp

```diff
@@ -184,36 +191,32 @@ namespace Babylon::Plugins
         {
             throw std::runtime_error{"Android Platform level < 24. Only camera texture override is available."};
         }
-
-        if (!overrideCameraTexture)
-        {
-            m_implData->cameraManager = GET_CAMERA_FUNCTION(ACameraManager_create)();
-        }
     }
 
     Camera::Impl::~Impl()
     {
         GET_CAMERA_FUNCTION(ACameraManager_delete)(m_implData->cameraManager);
     }
 
-    arcana::task<Camera::Impl::CameraDimensions, std::exception_ptr> Camera::Impl::Open(
-            std::shared_ptr<CameraDevice> cameraDevice, std::shared_ptr<CameraTrack> track)
+    arcana::task<Camera::Impl::CameraDimensions, std::exception_ptr> Camera::Impl::Open(CameraDevice cameraDevice, const CameraTrack* track)
```

Comment:
Does this need to be CameraDevice&?  I think this would call the copy constructor no?